### PR TITLE
Make Slide.url fall back to the linked event

### LINKED ISSUE #968: Automatically set referral event-slide hyperlink to event page
In GitLab by @JobDoesburg on Dec 4, 2019, 19:29

### One-sentence description

Automatically set referral event-slide hyperlink to event page 

### Motivation

Usability

### Desired functionality

If you set a slide to be an event slide, automatically fill in the hyperlink field with a URL to the event page (and maybe also auto fill in the title to be the event title?)

### PR DESCRIPTION
**Pull request by ColonelPhantom**
Closes #968 

### Summary
Slide.url is now a pseudo-property that checks either `custom_url` or `event`, and links to the event if `custom_url` is not set.

### How to test
Steps to test the changes you made:
1. Create a slide, give it an event but no link
2. View the homepage with the slide enabled
3. Observe that the slide now links to the relevant event

### CODE REVIEW

File: website/announcements/models.py [hunk 1]
```diff
@@ -242,5 +246,13 @@ def is_visible(self):
             and not (self.since is None and self.until is None)
         )
 
+    @property
+    def url(self):
+        if self.custom_url:
+            return self.custom_url
+        if self.event:
+            return reverse("events:event", kwargs={"pk": self.event.id})
```
Comment (DeD1rk) [inline website/announcements/models.py:254]: Can't we use event.get_absolute_url instead?

File: website/announcements/models.py [hunk 2]
```diff
@@ -188,9 +189,12 @@ class Slide(models.Model):
         verbose_name=_("Display only for authenticated members"), default=False
     )
 
-    url = models.URLField(
+    custom_url = models.URLField(
```
Comment (DeD1rk) [inline website/announcements/models.py:192]: I've got a hunch that this needs to be changed in the slide admin as well?

**Review by DeD1rk — CHANGES_REQUESTED** [review]

File: website/announcements/models.py (hunk 2 above)
Comment (ColonelPhantom) [reply website/announcements/models.py:192]: It seems the field is not listed there, SlideAdmin uses an implicit list of fields

File: website/announcements/models.py [hunk 3]
```diff
@@ -188,9 +188,12 @@ class Slide(models.Model):
...
         verbose_name=_("Link"),
-        help_text=_("Place the user is taken to when clicking the slide"),
+        help_text=_(
+            "Place the user is taken to when clicking the slide."
```
Comment (DeD1rk) [inline website/announcements/models.py:194]:
```suggestion
            "Place the user is taken to when clicking the slide. "
```

**Review by DeD1rk — APPROVED** [review]

**Comment by ColonelPhantom** [comment]
You'll need to recreate the migration if you're changing the help text.

**Comment by DeD1rk** [comment]
ouch

**Pull request merged**